Frends.AzureBlobStorage.UploadBlob/Frends.AzureBlobStorage.UploadBlob.Tests/UnitTestsAppend.cs

```diff
@@ -0,0 +1,473 @@
+            BlobType = AzureBlobType.Append,
+            ConnectionString = _connectionString,
+            CreateContainerIfItDoesNotExist = true,
+            FileEncoding = "utf-8",
+            HandleExistingFile = HandleExistingFile.Overwrite,
+            TenantID = null,
+            ApplicationID = null,
+            StorageAccountName = null,
+            ClientSecret = null,
+            ContentType = null,
+            PageMaxSize = default,
+            PageOffset = default,
+            ParallelOperations = default,
+            ResizeFile = default,
+        };
+
+        var destinationOA = new Destination
+        {
+            ConnectionMethod = ConnectionMethod.OAuth2,
+            ContainerName = _containerName,
+            BlobType = AzureBlobType.Append,
+            ConnectionString = _connectionString,
+            CreateContainerIfItDoesNotExist = true,
+            FileEncoding = "utf-8",
+            HandleExistingFile = HandleExistingFile.Overwrite,
+            TenantID = _tenantID,
+            ApplicationID = _appID,
+            StorageAccountName = _storageAccount,
+            ClientSecret = _clientSecret,
+            ContentType = null,
+            PageMaxSize = default,
+            PageOffset = default,
+            ParallelOperations = default,
+            ResizeFile = default,
+        };
+
+
+        var container = GetBlobContainer(_connectionString, _containerName);
+
+        // connection string
+        var result = await AzureBlobStorage.UploadBlob(input, destinationCS, options, default);
+        Assert.IsTrue(result.Success);
+        Assert.IsTrue(result.Data.ContainsValue($"{container.Uri}/compress.gz"));
+        Assert.IsTrue(await container.GetBlobClient("compress.gz").ExistsAsync(), "Uploaded SomeBlob blob should exist");
+
+        // OAuth
+        var result2 = await AzureBlobStorage.UploadBlob(input, destinationOA, options, default);
+        Assert.IsTrue(result2.Success);
+        Assert.IsTrue(result2.Data.ContainsValue($"{container.Uri}/compress.gz"));
+        Assert.IsTrue(await container.GetBlobClient("compress.gz").ExistsAsync(), "Uploaded SomeBlob blob should exist");
+    }
+
+    [TestMethod]
+    public async Task UploadFile_HandleExistingFile()
+    {
+        var errorHandlers = new List<HandleExistingFile>() { HandleExistingFile.Append, HandleExistingFile.Overwrite, HandleExistingFile.Error };
+        var options = new Options() { ThrowErrorOnFailure = true };
+
+        var input = new Source
+        {
+            SourceType = UploadSourceType.File,
+            SourceFile = _testfile,
+            Tags = default,
+            Compress = default,
+            ContentsOnly = default,
+            SearchPattern = default,
+            SourceDirectory = default,
+            BlobName = default,
+            BlobFolderName = default,
+        };
+
+        foreach (var handler in errorHandlers)
+        {
+            var destination = new Destination
```

Comment:
You could split it to 2 parts where most of the parameters have been set before foreach loop, but I just found it cleaner to  keep it like this. It's in the loop because of "HandleExistingFile = handler"

Reply:
Oh yeah. Missed that. But please split the Destination parameter setter in two because it makes the test more clear and limits the size of the file. You can the change the parameter's attribute inside the loop.